src/uu/sort/src/sort.rs

```diff
@@ -487,24 +514,39 @@ fn get_leading_number(a: &str) -> &str {
     return s;
 }
 
-// Matches GNU behavior, see:
-// https://www.gnu.org/software/coreutils/manual/html_node/sort-invocation.html
-// Specifically *not* the same as sort -n | uniq
-fn num_sort_dedup(a: &str) -> &str {
-    // Empty lines are dumped
-    if a.is_empty() {
-        return "0"
-    // And lines that don't begin numerically are dumped
-    } else if !a.trim().chars().nth(0).unwrap_or('\0').is_numeric() {
-        return "0"
-    } else {
+// For all dedups
+// Specifically *not* the same as sort | uniq
+// See: https://www.gnu.org/software/coreutils/manual/html_node/sort-invocation.html
+fn leading_num_numlines(a: &str) -> &str {
+    // Trim and remove any leading zeros
+    let s = a.trim().trim_start_matches('0');
+
+    // Get first char
+    let c = s.chars().nth(0).unwrap_or('\0');
+
+    // Empty, non-number lines, whitespace and are treated as same for dedup
+    if s.is_empty() {
+        return "";
```

Comment:
I'd get rid of the `return` statements here and let the value of the if statement be returned.

Reply:
Thanks for reviewing.  Absolutely.